encoding/rs/encode.go

```diff
@@ -127,3 +143,38 @@ func (g *Encoder) ExtendPolyEval(coeffs []fr.Element) ([]fr.Element, []fr.Elemen
 
 	return evals, pdCoeffs, nil
 }
+
+type JobRequest struct {
+	Index      uint64
+	FrameIndex uint64
+}
+
+func (g *Encoder) interpolyWorker(
+	polyEvals []fr.Element,
+	jobChan <-chan JobRequest,
```

Comment:
can it just use a workerpool, instead of a dedicated channel to create job request

Reply:
It is actually a good idea, I think we should create a global persistent object under prover? what do you think

Reply:
Say each requests is 256KB, 10MBps is equivalent to 40 parallel requests, and they need to share threads. If multiple objects are specified, it is unclear how to allocate them. I will mark it as todo, since I don't know the performance of the threadpool, compared to now.